### PR TITLE
dto validation done

### PR DESCRIPTION
**Pull request by khorkovenko**

### CODE REVIEW

**Review by Rommelua — CHANGES_REQUESTED** [review]
I can't believe that this validation works. Have you tested it?

File: store/src/main/java/com/book/store/dto/BookDto.java [hunk 1]
```diff
@@ -1,15 +1,31 @@
...
     private String title;
+
+    @NotBlank(message = "Author is required")
     private String author;
+
+    @NotBlank(message = "ISBN is required")
     private String isbn;
+
+    @NotNull(message = "Price is required")
+    @DecimalMin(value = "0.01", message = "Price must be greater than 0")
```
Comment (koval-kn) [inline store/src/main/java/com/book/store/dto/BookDto.java]:
```suggestion
    @DecimalMin(value = "0.0", inclusive = false, message = "Price must be greater than 0")
```

File: store/src/main/java/com/book/store/dto/BookDto.java [hunk 2]
```diff
@@ -1,15 +1,31 @@
...
     private Long id;
+
+    @NotBlank(message = "Title is required")
     private String title;
+
+    @NotBlank(message = "Author is required")
     private String author;
+
+    @NotBlank(message = "ISBN is required")
     private String isbn;
```
Comment (koval-kn) [inline store/src/main/java/com/book/store/dto/BookDto.java:14]:
```suggestion
    @Size(min = 1, max = 13, message = "ISBN must be between 1 and 13 characters.")
    private String isbn;
```

File: store/src/main/java/com/book/store/dto/BookDto.java [hunk 3]
```diff
@@ -1,15 +1,31 @@
...
 
 @Data
 public class BookDto {
     private Long id;
+
+    @NotBlank(message = "Title is required")
     private String title;
+
+    @NotBlank(message = "Author is required")
     private String author;
```
Comment (koval-kn) [inline store/src/main/java/com/book/store/dto/BookDto.java:12]:
```suggestion
    @Size(min = 1, max = 255, message = "Author must be between 1 and 255 characters.")
    private String author;
```

File: store/src/main/java/com/book/store/dto/BookDto.java [hunk 4]
```diff
@@ -1,15 +1,31 @@
...
+import jakarta.validation.constraints.NotNull;
 import java.math.BigDecimal;
 import lombok.Data;
 
 @Data
 public class BookDto {
     private Long id;
+
+    @NotBlank(message = "Title is required")
     private String title;
```
Comment (koval-kn) [inline store/src/main/java/com/book/store/dto/BookDto.java:10]:
```suggestion
    @Size(min = 1, max = 255, message = "Title must be between 1 and 255 characters.")
    private String title;
```

File: store/src/main/java/com/book/store/dto/BookDto.java [hunk 5]
```diff
@@ -1,15 +1,31 @@
...
+
+    @NotBlank(message = "ISBN is required")
     private String isbn;
+
+    @NotNull(message = "Price is required")
+    @DecimalMin(value = "0.01", message = "Price must be greater than 0")
     private BigDecimal price;
+
+    @NotBlank(message = "Description is required")
     private String description;
```
Comment (koval-kn) [inline store/src/main/java/com/book/store/dto/BookDto.java:18]:
```suggestion
    @Size(max = 2000, message = "Description can't be longer than 2000 characters.")
    private String description;
```

File: store/src/main/java/com/book/store/dto/BookRequestDto.java [hunk 6]
```diff
@@ -1,14 +1,29 @@
...
 
+import jakarta.validation.constraints.DecimalMin;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.NotNull;
 import java.math.BigDecimal;
 import lombok.Data;
 
 @Data
 public class BookRequestDto {
+    @NotBlank(message = "Title is required")
```
Comment (koval-kn) [inline store/src/main/java/com/book/store/dto/BookRequestDto.java:13]: same in this class

**Review by koval-kn — CHANGES_REQUESTED** [review]

File: store/src/main/java/com/book/store/dto/BookDto.java [hunk 7]
```diff
@@ -1,15 +1,37 @@
...
 import java.math.BigDecimal;
+
+import jakarta.validation.constraints.Size;
 import lombok.Data;
 
 @Data
 public class BookDto {
     private Long id;
+
+    @NotBlank(message = "Title is required")
```
Comment (sokimaaa) [inline store/src/main/java/com/book/store/dto/BookDto.java]: Have you used BookDto validations? I didnt see

Code that are not in use should be removed

**Pull request merged**